context/src/main/java/io/grpc/ThreadLocalContextStorage.java

```diff
@@ -28,11 +28,18 @@
   /**
    * Currently bound context.
    */
-  private static final ThreadLocal<Context> localContext = new ThreadLocal<Context>();
+  private static final ThreadLocal<Context> localContext = new ThreadLocal<Context>() {
+    @Override
+    protected Context initialValue() {
+      return Context.ROOT;
```

Comment:
This would likely bite any implementation, because current and attach are working differently. I see two options:

1. We remove the `if (current == null)` check in `current()` and require the storage to handle that case. We would still want a `current == null` check, but it would throw an exception instead of choosing ROOT.

2. Have `Context.attach()` check the return and use `ROOT` if it is `null`. This is functionally equivalent to what was being done before.